services/frontend-service/src/ui/ActionDetails.tsx

```diff
@@ -82,104 +118,181 @@ type ActionDetails = {
+                action.createApplicationLock.application +
+                '" on ' +
+                action.createApplicationLock.environment,
+            icon: <LockRounded />,
+            environment: action.createApplicationLock.environment,
+            application: action.createApplicationLock.application,
+        };
+    if ('deleteEnvironmentLock' in action)
+        return {
+            type: ActionTypes.DeleteEnvironmentLock,
+            name: 'Delete Env Lock',
+            dialogTitle: 'Are you sure you want to delete this environment lock?',
+            summary: 'Delete environment lock on ' + action.deleteEnvironmentLock.environment,
+            icon: <LockOpenRounded />,
+            environment: action.deleteEnvironmentLock.environment,
+            lockId: action.deleteEnvironmentLock.lockId,
+        };
+    if ('deleteApplicationLock' in action)
+        return {
+            type: ActionTypes.DeleteApplicationLock,
+            name: 'Delete App Lock',
+            dialogTitle: 'Are you sure you want to delete this application lock?',
+            summary:
+                'Unlock "' +
+                action.deleteApplicationLock.application +
+                '" on ' +
+                action.deleteApplicationLock.environment,
+            icon: <LockOpenRounded />,
+            environment: action.deleteApplicationLock.environment,
+            application: action.deleteApplicationLock.application,
+            lockId: action.deleteApplicationLock.lockId,
+        };
+    if ('prepareUndeploy' in action)
+        return {
+            type: ActionTypes.PrepareUndeploy,
+            name: 'Prepare Undeploy',
+            dialogTitle: 'Are you sure you want to start undeploy?',
+            description:
+                'The new version will go through the same cycle as any other versions' +
+                ' (e.g. development->staging->production). ' +
+                'The behavior is similar to any other version that is created normally.',
+            summary: 'Prepare undeploy version for Application ' + action.prepareUndeploy.application,
+            icon: <DeleteOutlineRounded />,
+            application: action.prepareUndeploy.application,
+        };
+    if ('undeploy' in action)
+        return {
+            type: ActionTypes.Undeploy,
+            name: 'Undeploy',
+            dialogTitle: 'Are you sure you want to undeploy this application?',
+            description: 'This application will be deleted permanently',
+            summary: 'Undeploy and delete Application ' + action.undeploy.application,
+            icon: <DeleteForeverRounded />,
+            application: action.undeploy.application,
+        };
+    else
+        return {
+            type: ActionTypes.UNKNOWN,
+            name: 'invalid',
+            dialogTitle: 'invalid',
+            summary: 'invalid',
+            icon: <Error />,
+        };
+};
+
+const randomLockId = () => 'ui-' + Math.random().toString(36).substring(7);
```

Comment:
I had to look up what `Math.random().toString(36).substring(7)` does. Having a named `const randBase36 = () => Math.random().toString(36).substring(7)` function, with a hint that the result will match `[0-9A-Z]{7}` and a link to https://en.wikipedia.org/wiki/Base36, might have helped.